QSPIFBlockDevice.cpp

```diff
@@ -0,0 +1,1263 @@
+    bool memReady = true;
+
+    do {
+        retries++;
+        //Read the Status Register from device
+        if (QSPI_STATUS_OK != _qspi_send_general_command(QSPIF_RDSR, -1, NULL, 0, status_value,
+                2)) {   // store received values in status_value
+            QSPIF_LOG(QSPIF_DEBUG_ERROR, "ERROR: Reading Status Register failed\n");
+        }
+    } while ( (status_value[0] & 0x1) != 0 && retries < IS_MEM_READY_MAX_RETRIES );
+
+    if ((status_value[0] & 0x1) != 0) {
+        QSPIF_LOG(QSPIF_DEBUG_DEBUG, "ERROR: _is_mem_ready FALSE\n");
+        memReady = false;
+    }
+    return memReady;
+}
+
+
+int QSPIFBlockDevice::_set_write_enable()
+{
+    int status = 0;
+    if (QSPI_STATUS_OK !=  _qspi_send_general_command(QSPIF_WREN, -1, NULL, 0, NULL, 0)) {
+        QSPIF_LOG(QSPIF_DEBUG_ERROR, "ERROR:Sending WREN command FAILED\n");
+        status = -1;
+    }
+    return status;
+}
+
+
+
+/*********************************************/
+/************* Utility Functions *************/
+/*********************************************/
+int QSPIFBlockDevice::_utils_find_addr_region(int offset)
+{
+    //Find the region to which the given offset belong to
+    if ((offset > (int)_device_size_bytes) || (_regions_count == 0)) {
+        return -1;
+    }
+
+    if (_regions_count == 1) {
+        return 0;
+    }
+
+    for (int i_ind = _regions_count - 2; i_ind >= 0; i_ind--) {
+
+        if (offset > _region_high_boundary[i_ind]) {
+            return (i_ind + 1);
+        }
+    }
+    return -1;
+
+}
+
+
+int QSPIFBlockDevice::_utils_iterate_next_largest_erase_type(uint8_t& bitfield, int size, int offset, int boundry)
+{
+    // Iterate on all supported Erase Types of the Region to which the offset belong to.
+    // Iterates from highest type to lowest
+    uint8_t type_mask = ERASE_BITMASK_TYPE4;
+    int i_ind  = 0;
+    int largestEraseType = 0;
+    for (i_ind = 3; i_ind >= 0; i_ind--) {
+        if (bitfield & type_mask) {
+            largestEraseType = i_ind;
+            if ( (size > _erase_type_size_arr[largestEraseType]) &&
+                    ((boundry - offset) > _erase_type_size_arr[largestEraseType]) ) {
+                break;
+            } else {
+                bitfield &= ~type_mask;
+            }
+        }
+        type_mask = type_mask >> 1;
+    }
+
+    if (i_ind == 4) {
+        QSPIF_LOG(QSPIF_DEBUG_ERROR, "ERROR: no erase type was found for current region addr");
+    }
+    return largestEraseType;
+
+}
+
+
+/***************************************************/
+/*********** QSPI Driver API Functions *************/
+/***************************************************/
+
+qspi_status_t QSPIFBlockDevice::_qsp_set_frequency(int freq)
+{
+    return _qspi.set_frequency(freq);
+}
+
+
+qspi_status_t QSPIFBlockDevice::_qspi_send_read_command(unsigned int read_inst, void *buffer, bd_addr_t addr,
+        bd_size_t size)
+{
+    // Send Read command to device driver
+    size_t bufLen = size;
```

Comment:
`buf_len` , same as `read_inst` in this scope

Reply:
fixed